src/nvim/version.c

```diff
@@ -2715,8 +2715,9 @@ void list_version(void)
 /// Whether it still is not too late to show an intro message
 bool may_show_intro(void)
 {
-  return (buf_is_empty(firstwin->w_buffer)
-          && (firstwin->w_buffer->b_fname == NULL)
+  return (buf_is_empty(curbuf)
+          && (curbuf->b_fname == NULL)
+          && (curbuf->handle == 1)
```

Comment:
This will show intro if buffer 1 is shown in a floating window that has been entered, regardless of what the buffer in the non-floating window is.

Reply:
Good point.
I guess the proper approach would be to also clear intro if current window is not the one as it was during start? Adding `&& (curwin->handle == 1000)` condition works. Is this an appropriate solution here?

Reply:
This should be resolved now.